### PR TITLE
Issue #876: Updated code for finding variants.

### PR DESCRIPTION
**Pull request by susanodd**
This issue revises the `find_and_save_variants` code in `views.py`.

- accommodates datasets and selected datasets
- non Latin languages (e.g., Japanese)
- makes use of a template (used to have hard-coded html)
- identifies focus glosses with annotations of pattern ending in -A, -B, ...
- displays a table showing other glosses that end in a different letter that are not related in some other way
- does not update anything, merely displays the patterns it finds (it used to save variants as new relations)

The code for `get_stems` (the pattern before the suffix) has been flattened out in order to work with non-Latin character sets.
The list comprehension did not work on Japanese.
See screenshots in the issue.
The code for actually saving the found variants has been commented. This can be added as a button after getting feedback on usage. Originally it was foreseen that this code would be run in a cron job.

### CODE REVIEW

**Review by vanlummelhuizen — COMMENTED** [review]
@susanodd Nice work! Nice functionality! I have some minor code issues, which you can easily work out, I'm sure.

File: signbank/dictionary/views.py [hunk 1]
```diff
@@ -2338,117 +2338,102 @@ def add_handshape_image(request):
...
+        else:
+            translations.append("{}".format(translation.text))
+    return ", ".join(translations)
 
-        dict_key = int(gloss.id)
-        gloss_pattern_table[dict_key] = '<td>' + str(gloss.idgloss) + '</td>'
-        other_relations_of_sign = gloss.other_relations()
 
-        if other_relations_of_sign:
+def find_and_save_variants(request):
```
[the remark this answers is not in the document]
Comment (susanodd) [reply signbank/dictionary/views.py:2357]: > Some function documentation?

Hmmm. For this one, it's easier said than done.

File: signbank/dictionary/views.py [hunk 2]
```diff
@@ -2338,117 +2338,102 @@ def add_handshape_image(request):
...
-    gloss_table_suffix = '</tbody>\n' \
-                         '</table>\n' \
-                         '</body>\n' \
-                         '</html>'
-
-    gloss_pattern_table = dict()
-    gloss_table_rows = ''
-
-    for gloss in variant_pattern_glosses:
+def gloss_annotations(this_gloss):
```
[the remark this answers is not in the document]
Comment (susanodd) [reply signbank/dictionary/views.py:2345]: > Some function documentation?

Done.

**Comment by susanodd** [comment]
The requested improvements have been made!

**Comment by susanodd** [comment]
@vanlummelhuizen this is complete. Can it be merged?
I made a new issue for the comprehension rewrite. That was originally the reason I requested @Woseseltops because that seems like down-grading to remove a comprehension. I explained why in the issue #890

**Pull request merged**